frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/block-pool-modal.tsx

```diff
@@ -0,0 +1,147 @@
+      namespaced: true,
+      isList: false,
+      name: state.poolName,
+      namespace: CEPH_STORAGE_NAMESPACE,
+    };
+  }, [state.poolName]);
+
+  const [newPool, newPoolLoaded, newPoolLoadError] = useK8sWatchResource<StoragePoolKind>(
+    poolResource,
+  );
+
+  React.useEffect(() => {
+    if (isSubmit) {
+      if (newPool && newPoolLoaded && newPool?.status?.phase === POOL_STATE.READY) {
+        dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.CREATED });
+        setIsSubmit(false);
+        clearTimeout(timer);
+      } else if (newPoolLoaded && newPool?.status?.phase === POOL_STATE.FAILED) {
+        dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.FAILED });
+        setIsSubmit(false);
+        clearTimeout(timer);
+      } else if (newPoolLoaded && newPoolLoadError && newPoolLoadError?.response?.status !== 404) {
+        dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.FAILED });
+        setIsSubmit(false);
+        clearTimeout(timer);
+      }
+    }
+  }, [isSubmit, newPool, newPoolLoadError, newPoolLoaded, timer]);
+
+  // Create new pool
+  React.useEffect(() => {
+    if (state.isSubmitted) {
+      dispatch({ type: BlockPoolActionType.SET_IS_SUBMITTED, payload: false });
+      dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.PROGRESS });
+      const poolObj: StoragePoolKind = getPoolKindObj(state);
+
+      handlePromise(
+        k8sCreate(CephBlockPoolModel, poolObj),
+        () => {
+          setIsSubmit(true);
+          // The modal will wait for 15 sec to get feedback from Rook
+          const timeoutTimer = setTimeout(() => {
+            dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.TIMEOUT });
+            setIsSubmit(false);
+          }, 30 * SECOND);
+          setTimer(timeoutTimer);
+        },
+        () => {
+          dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.FAILED });
+        },
+      );
+    }
+  }, [handlePromise, state, state.isSubmitted]);
+
+  return (
+    <div className="modal-content modal-content--no-inner-scroll">
+      <ModalTitle>{MODAL_TITLE}</ModalTitle>
+      <ModalBody>
+        <p>{MODAL_DESC}</p>
+        {state.poolStatus ? (
+          <div key="progress-modal">
+            <BlockPoolStatus status={state.poolStatus} name={state.poolName} error={errorMessage} />
+          </div>
+        ) : (
+          <BlockPoolBody
+            cephCluster={cephClusterObj[0]}
+            state={state}
+            dispatch={dispatch}
+            showPoolStatus
+          />
+        )}
+      </ModalBody>
+      <ModalFooter inProgress={state.poolStatus === POOL_PROGRESS.PROGRESS}>
+        <BlockPoolModalFooter
+          poolName={state.poolName}
+          poolStatus={state.poolStatus}
+          replicaSize={state.replicaSize}
+          volumeType={state.volumeType}
```

Comment:
Why not pass a state and not pass each property differently

Reply:
ack

Reply:
ack

Reply:
@afreen23 now, i am using most of the state var in footer, so i am passing state directly

Reply:
ok